src/cryptonote_core/cryptonote_tx_utils.cpp

```diff
@@ -306,11 +284,64 @@ namespace cryptonote
     uint64_t summary_outs_money = 0;
     //fill outputs
     size_t output_index = 0;
+
+    // figure out if we need to make additional tx pubkeys
+    size_t num_stdaddresses = 0;
+    size_t num_subaddresses = 0;
+    std::unordered_set<cryptonote::account_public_address> unique_dst_addresses;
+    for (const auto& i : destinations)
+    {
+      if (i.addr == change_addr)
+        continue;
+      if (unique_dst_addresses.count(i.addr) == 0)
+      {
+        unique_dst_addresses.insert(i.addr);
+        if (i.is_subaddress)
+          ++num_subaddresses;
+        else
+          ++num_stdaddresses;
+      }
+    }
+
+    cryptonote::account_public_address dst_subaddr_with_shared_txpubkey;
+    // if all the destinations are subaddresses, we can pick one of them and use the shared tx pubkey for it
+    if (num_stdaddresses == 0)
+    {
+      do
+      {
+        dst_subaddr_with_shared_txpubkey = shuffled_dsts[crypto::rand<size_t>() % shuffled_dsts.size()].addr;
```

Comment:
But I see `destinations` (and thus `shuffled_dsts` also) is ensured to be non-empty at L164.